locker/control_plane_init_locker.go

```diff
@@ -27,28 +27,15 @@ import (
 	clusterv2 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha2"
 )
 
-// ControlPlaneInitLocker provides a locking mechanism for cluster initialization.
-type ControlPlaneInitLocker interface {
-	// Acquire returns true if it acquires the lock for the cluster.
-	Acquire(cluster *clusterv2.Cluster) bool
-}
-
-// controlPlaneInitLocker uses a ConfigMap to synchronize cluster initialization.
-type controlPlaneInitLocker struct {
+// ControlPlaneInitLocker uses a ConfigMap to synchronize cluster initialization.
+type ControlPlaneInitLocker struct {
 	log             logr.Logger
 	configMapClient corev1.ConfigMapsGetter
 }
 
-var _ ControlPlaneInitLocker = &controlPlaneInitLocker{}
-
-func newControlPlaneInitLocker(log logr.Logger, configMapClient corev1.ConfigMapsGetter) *controlPlaneInitLocker {
-	return &controlPlaneInitLocker{
-		log:             log,
-		configMapClient: configMapClient,
-	}
-}
-
-func (l *controlPlaneInitLocker) Acquire(cluster *clusterv2.Cluster) bool {
+// Acquire will attempt to acquire a lock that guarantees this is the first control plane to be created.
+// TODO Consider adding a short circuit if there is already a control plane node to reduce the calls to the API server.
```

Comment:
What kind of short circuit did you have in mind? This code does a "get" check first. Were you thinking of something else?

Reply:
like, if there is already a control plane node we don't need to worry about locks any more since control planes can be joined in parallel as well as machines. But that is very likely a premature optimization.

Reply:
Given the discussion in https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/42#discussion_r310661445, we may no longer need the locker code. TBD. WDYT?